jbmc/regression/jbmc-strings/StringToLowerCase/Test.java

```diff
@@ -0,0 +1,50 @@
+public class Test {
+    public String det()
+    {
+        StringBuilder builder = new StringBuilder();
+        builder.append("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ".toLowerCase());
+        builder.append("abcdeghijklmnopfrstuqwxyzABCDvFGHIJKLMENOPQRSTUVWXYZ".toLowerCase());
```

Comment:
do you have tests for the other letters too?

Reply:
No, but we don't test the result here, we are just testing that unnecessary constraints are not added. I didn't find a good way to test it here: we cannot add assertions because that could change the result and recovering the string from the trace is difficult.

Reply:
A general, unhelpful point -- are we doing ASCII case conversion or Unicode case conversion, because the later, frankly, fills me with fear:

https://unicode.org/faq/casemap_charprop.html
https://www.unicode.org/reports/tr21/tr21-5.html

Reply:
I seem to recall seeing tests for upper-casing non-ascii characters...